.github/workflows/docker.yml

```diff
@@ -1,4 +1,26 @@
 name: "Docker Build"
+env:
+  repository: "cloudcustodian"
+  job_metadata: '
+    {
+      "c7n": {
+        "name": "cli",
```

Comment:
afaics we just need to rename cli docker file to c7n, and drop using dockerpkg.py more fully, and then we can cut out all the redundant variables here (image_var, name, image), and also get rid of these inline blocks of json here and simply the matrix job as well.